sklearn/model_selection/_split.py

```diff
@@ -1769,6 +1769,31 @@ def __init__(self, *, n_splits=5, n_repeats=10, random_state=None):
             n_splits=n_splits,
         )
 
+    def split(self, X, y, groups=None):
+        """Generate indices to split data into training and test set.
+
+        Parameters
+        ----------
+        X : array-like of shape (n_samples, n_features)
+            Training data, where `n_samples` is the number of samples
+            and `n_features` is the number of features.
+
+        y : array-like of shape (n_samples,), y parameter is required
+            The target variable for supervised learning problems.
+
+        groups : object
+            Always ignored, exists for compatibility.
+
+        Yields
+        ------
+        train : ndarray
+            The training set indices for that split.
+
+        test : ndarray
+            The testing set indices for that split.
+        """
+        return super().split(X, y, groups=groups)
```

Comment:
Sorry I spoke too soon I think you should model this after `StratifiedKFold.split`: https://github.com/scikit-learn/scikit-learn/blob/64ab789905077ba8990522688c11177442e5e91f/sklearn/model_selection/_split.py#L850-L856

Reply:
Ok, should I add this extra check:

```python
y = check_array(y, input_name="y", ensure_2d=False, dtype=None) 
```


And, this below one is already being checked in _UnsupportedGroupCVMixin.split()

```python
 if groups is not None: 
     warnings.warn( 
         f"The groups parameter is ignored by {self.__class__.__name__}", 
         UserWarning, 
     ) 
```

Reply:
> Ok, should I add this extra check

Yes I think this makes sense to check `y` the same way that `StratifiedKFold` does.

> And, this below one is already being checked in _UnsupportedGroupCVMixin.split()

This is what I thought originally as well, not sure now with the multiple inheritance going on. This should be not too hard to check that the warning is issued with a code snippet.

Reply:
When calling split function, it is being used from _UnsupportedGroupCVMixin due to its first in order in multiple inheritance. 

So not required to check  `if groups is not None`

Sending the updated commit which the new changes.